contracts/core/PositionVault.sol

```diff
@@ -316,43 +294,140 @@ contract PositionVault is Constants, ReentrancyGuard, IPositionVault {
             order.positionType = POSITION_LIMIT;
             order.lmtPrice = _params[0];
         } else if (_orderType == OrderType.STOP) {
+            require(_params[1] > 0, "stop price is invalid");
+
             order.status = OrderStatus.PENDING;
             order.positionType = POSITION_STOP_MARKET;
             order.stpPrice = _params[1];
         } else if (_orderType == OrderType.STOP_LIMIT) {
+            require(_params[0] > 0 && _params[1] > 0, "stop limit price is invalid");
+
             order.status = OrderStatus.PENDING;
             order.positionType = POSITION_STOP_LIMIT;
             order.lmtPrice = _params[0];
             order.stpPrice = _params[1];
         }
+
         lastPosId += 1;
         emit NewOrder(_account, _indexToken, _isLong, lastPosId - 1, order.positionType, order.status, _params);
     }
 
+    function executeOpenMarketOrder(uint256 _posId) public nonReentrant {
+        require(settingsManager.isManager(msg.sender), "You are not allowed to trigger");
+
+        Position memory position = positions[_posId];
+        Order storage order = orders[_posId];
+        require(order.size > 0, "order size is 0");
+        require(order.positionType == POSITION_MARKET, "not market order");
+        require(order.status == OrderStatus.PENDING, "not pending order");
+
+        settingsManager.updateFunding(position.indexToken);
+
+        uint256 fee = settingsManager.collectMarginFees(
+            position.owner,
+            position.indexToken,
+            position.isLong,
+            order.size
+        );
+        uint256 price = priceManager.getLastPrice(position.indexToken);
+        checkSlippage(position.isLong, order.lmtPrice, price);
+
+        _increasePosition(
+            position.owner,
+            position.indexToken,
+            order.collateral + fee,
+            order.size,
+            lastPosId,
+            price,
+            position.isLong
+        );
+
+        order.collateral = 0;
+        order.size = 0;
+        order.status = OrderStatus.FILLED;
+        _addUserAlivePosition(position.owner, lastPosId);
+
+        emit UpdateOrder(_posId, order.positionType, order.status);
+    }
+
+    function _executeOpenMarketOrders(uint256 numOfOrders) internal {
+        uint256 index = openMarketQueueIndex;
+        uint256 endIndex = index + numOfOrders;
+        uint256 length = openMarketQueuePosIds.length;
+
+        if (index >= length) return;
+        if (endIndex > length) endIndex = length;
+
+        while (index < endIndex) {
+            uint256 posId = openMarketQueuePosIds[index];
+
+            try this.executeOpenMarketOrder(posId) {} catch Error(string memory err) {
+                cancelMarketOrder(posId);
+                emit MarketOrderExecutionError(posId, positions[posId].owner, err);
+            } catch (bytes memory) {
+                cancelMarketOrder(posId);
+            }
+
+            delete openMarketQueuePosIds[posId];
```

Comment:
CHANGE: should be `delete openMarketQueuePosIds[index];` ?

Reply:
correct, nice finding

Reply:
correct, nice find!